tests/unit/lms/services/oauth2_token_test.py

```diff
@@ -43,7 +42,7 @@ def test_get_raises_OAuth2TokenError_with_no_token(
     ):
         wrong_value = "WRONG"
         if wrong_param == "application_instance":
-            application_instance.consumer_key = "WRONG"
+            application_instance.id = 100
             wrong_value = application_instance
```

Comment:
I think this test might be a case of superficial similarity. The code used to try and account for the difference between the two cases is about the same as just having two cases. e.g.:

```python
    def test_get_raises_OAuth2TokenError_with_mismatching_application_instance(
        self, db_session, application_instance, lti_user
    ):
        service = OAuth2TokenService(
            db_session,
            application_instance=factories.ApplicationInstance(),
            user_id=lti_user.user_id,
        )

        with pytest.raises(OAuth2TokenError):
            service.get()

    def test_get_raises_OAuth2TokenError_with_mismatching_user(
        self, db_session, application_instance, lti_user
    ):
        service = OAuth2TokenService(
            db_session, application_instance=application_instance, user_id="WRONG"
        )

        with pytest.raises(OAuth2TokenError):
            service.get()
```

If we want to get very fancy, we can use a partial to create the service. This has the advantage of only including the field that matters in the definition of the test, which makes it clearer what's changing / important:

```python
    from functools import partial

   ...

    def test_get_raises_OAuth2TokenError_with_bad_ai(self, get_service):
        service = get_service(application_instance=factories.ApplicationInstance())

        with pytest.raises(OAuth2TokenError):
            service.get()

    def test_get_raises_OAuth2TokenError_with_bad_user(self, get_service):
        service = get_service(user_id="WRONG")

        with pytest.raises(OAuth2TokenError):
            service.get()

    ...

    @pytest.fixture
    def svc(self, get_service):
        return get_service()

    @pytest.fixture
    def get_service(self, pyramid_request, application_instance):
        return partial(
            OAuth2TokenService,
            db=pyramid_request.db,
            application_instance=application_instance,
            user_id=pyramid_request.lti_user.user_id,
        )
```

You could also skip the `svc` fixture here and use `get_service` in every test without much loss for even shorter tests:

```python
    def test_get_returns_token_when_present(self, get_service, oauth_token):
        assert get_service().get() == oauth_token

    def test_get_raises_OAuth2TokenError_with_bad_ai(self, get_service):
        with pytest.raises(OAuth2TokenError):
            get_service(application_instance=factories.ApplicationInstance()).get()

    def test_get_raises_OAuth2TokenError_with_bad_user(self, get_service):
        with pytest.raises(OAuth2TokenError):
            get_service(user_id="WRONG").get()
```

Where your appetite falls off for compactness may vary. I think keeping the arrange / act / assert setup is probably worth it.